.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,27 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+	"name": "Existing Dockerfile",
+	"build": {
+		// Sets the run context to one level up instead of the .devcontainer folder.
+		"context": "..",
+		// Update the 'dockerFile' property if you aren't using the standard 'Dockerfile' filename.
+		"dockerfile": "../Dockerfile"
```

Comment:
Is it necessary to have the `Dockerfile` above the Dev Container configuration? I've often seen the `Dockerfile` of a Dev Container in the `.devcontainer` directory.

It might be a bit confusing to see the `Dockerfile` on its own as it doesn't/can't run/build the MkDocs website out-of-the-box.

Reply:
The docker file can run the docs with a docker-compose with the command argument. This is quite handy for people that do not use VSCode. I thought about putting it into a `docker/` folder at the root. This would not make it specific to `.devcontainer`. But I think having it in the root is not the best option. So what do you think, in `.devcontainer` or a new `docker/` folder?

Reply:
OK. In this case, I would move the `.devcontainer/devcontainer.json` file to `.devcontainer.json` (move it and rename it at the root level of the Git repository).

Then, I would leave the `Dockerfile` but I would add a `docker-compose.yml` file at the root level of the Git repository. So people using VSCode would use the devcontainer configuration and others would use the docker-compose.yml configuration.

The Git repository would look like this:

```
.github/
docs/
infrastructure/
.devcontainer.json
.gitignore
.gitlab-ci.yml
Dockerfile
docker-compose.yml
README.md
mkdocs.yml
```

What do you think?

Reply:
Sounds good!